File: tests/fixtures/check_then_act/ts/unrelated_member_recheck_fixture.ts
// A member flag checked after the await says nothing about the row read before it.
export class Worker {
  private closed = false;
  constructor(private readonly repo: any) {}

  async run(id: string) {
    const row = this.repo.get(id);
    if (row) return; // expect: check
    await this.repo.sync(id); // expect: gap await
    if (this.closed) return;
    this.repo.update(id); // expect: write
  }
}

File: tests/fixtures/check_then_act/py/unrelated_member_recheck_fixture.py
# A member flag checked after the await says nothing about the row read before it.
class Worker:
    def __init__(self, repo):
        self.closed = False
        self.repo = repo

    async def run(self, item_id):
        row = self.repo.get(item_id)
        if row:  # expect: check
            return
        await self.repo.sync(item_id)  # expect: gap await
        if self.closed:
            return
        self.repo.update(item_id)  # expect: write
